src/components/ElementSearchDialog/element-search-dialog.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { ReactNode, useCallback, useMemo } from 'react';
+import {
+    Autocomplete,
+    Dialog,
+    DialogContent,
+    TextField,
+    AutocompleteInputChangeReason,
+    AutocompleteChangeReason,
+} from '@mui/material';
+import { Search, SearchOff } from '@mui/icons-material';
+import { useIntl } from 'react-intl';
+import { EquipmentInfos } from '../../index';
+import * as React from 'react';
+
+interface ElementSearchDialogProps {
+    open: boolean;
+    onClose: () => void;
+    searchingLabel?: string;
+    searchTerm: string;
+    onSearchTermChange: (searchTerm: string) => void;
+    onSelectionChange: (selection: EquipmentInfos) => void;
+    elementsFound: EquipmentInfos[]; // [{ label: aLabel, id: anId }, ...]
+    renderElement: (props: any) => ReactNode;
+    searchTermDisabled?: boolean;
+    searchTermDisableReason?: string;
+    isLoading: boolean;
+}
+
+const ElementSearchDialog = (props: ElementSearchDialogProps) => {
+    const intl = useIntl();
+
+    const {
+        open,
+        onClose,
+        searchingLabel,
+        searchTerm,
+        onSearchTermChange,
+        onSelectionChange,
+        elementsFound, // [{ label: aLabel, id: anId }, ...]
+        renderElement,
+        searchTermDisabled,
+        searchTermDisableReason,
+        isLoading,
```

Comment:
Why do we need to expose this prop to the owner of `ElementSearchDialog` ?
Maybe we could set a local state to true when `onSearchTermChange` is called then set it to false when elementsFound changes ?
I think It's complex if we let the component owner manage this state no ?
At least name is `loading` as the Autocomplete prop ?

Reply:
I think it's way better to handle "isLoading" when we make the fetch, otherwise we will have some troubles handling errors or edge cases

for "isLoading" naming, I see your point, but I think it's more clear to call it "isLoading"
Can we ask a 3rd person to decide with us ?

Reply:
?on which case we can have troubles or errors when using state instead of props